keycache/keycache.go

```diff
@@ -78,76 +88,81 @@ func (usage Usage) matches(user string, labels []string) bool {
 }
 
 func NewCache() Cache {
-	return Cache{make(map[string]ActiveUser)}
+	return Cache{make(map[DelegateIndex]ActiveUser)}
 }
 
 // setUser takes an ActiveUser and adds it to the cache.
-func (cache *Cache) setUser(in ActiveUser, name string) {
-	cache.UserKeys[name] = in
+func (cache *Cache) setUser(in ActiveUser, name, slot string) {
+	d := &DelegateIndex{Name: name, Slot: slot}
+	cache.UserKeys[*d] = in
 }
 
 // Valid returns true if matching active user is present.
 func (cache *Cache) Valid(name, user string, labels []string) (present bool) {
-	key, present := cache.UserKeys[name]
-	if present {
+	for d, key := range cache.UserKeys {
+		if d.Name != name {
+			continue
+		}
 		if key.Usage.matches(user, labels) {
 			return true
-		} else {
-			present = false
 		}
 	}
 
-	return
+	return false
 }
 
 // matchUser returns the matching active user if present
 // and a boolean to indicate its presence.
-func (cache *Cache) matchUser(name, user string, labels []string) (out ActiveUser, present bool) {
-	key, present := cache.UserKeys[name]
-	if present {
+func (cache *Cache) matchUser(name, user string, labels []string) (out ActiveUser, slot string, present bool) {
```

Comment:
If the variables aren't going to be used, we should remove them from the function signature.